articles/content_filtering.md

```diff
@@ -0,0 +1,315 @@
+  "uuid = %0 OR uuid = %1" DEADBEEF CAFEFEED : subscribe only interest goal ids which client handlers possess.
+  ```
+- Multiple filter_expression and expression_parameters can be modified dynamically at runtime.
+  This is because of use cases for `/parameter_events` and action `feedback` and `status` topics, parameter filtering expression is dependent on user application, and action client has multiple goal id to handle.
+
+### Improvement Result
+
+This result indicates if ContentFilteredTopic provides the improvement for CPU consumption and network traffic as we described above. Using single writer and multiple readers up to 10 with filtering expression and expression parameters.
+
+#### Environment
+
+- Intel(R) Core(TM) i7-7700 CPU @ 3.60GHz / 8GB memory
+- Ubuntu 18.04.5 LTS
+- docker 19.03.6
+- ubuntu:20.04 container
+- ros2:rolling, rti-connext-dds-5.3.1
+
+#### IDL
+
+```
+struct cft {
+    long count;
+    string flag;
+    string cmd;
+    long data_size;
+    string<8 * 1024 * 1024> data;
+};
+```
+
+#### Result
+
+`flag` means how many **readers** setup with filtering parameter `yes`. (means that receive will receive the message) `no filter` means in `normal topic` w/o content filtering. `improvement rate` means how much network traffic bytes to be saved compare to `no filter` mode, below is the formula:
+
+![formula](../img/content_filter/formula.png)
+
+- Publication Frequency: 10Hz, Data Size: 1KByte, Count: 3000 times
+
+| flags | cpu | memory | tx (Network load) | improvement |
+| --- | --- | --- | --- | --- |
+| 1/10 'yes' | 0.40% | 0.2% | 11.49 KByte/s | 90% |
+| 2/10 'yes' | 0.46% | 0.2% | 22.9 KByte/s | 80% |
+| 3/10 'yes' | 0.48% | 0.2% | 34.28 KByte/s | 70% |
+| 4/10 'yes' | 0.55% | 0.2% | 45.69 KByte/s | 60% |
+| 5/10 'yes' | 0.60% | 0.2% | 57.09 KByte/s | 50% |
+| 6/10 'yes' | 0.69% | 0.2% | 68.49 KByte/s | 40% |
+| 7/10 'yes' | 0.73% | 0.2% | 79.89 KByte/s | 30% |
+| 8/10 'yes' | 0.79% | 0.2% | 91.29 KByte/s | 20% |
+| 9/10 'yes' | 0.84% | 0.2% | 102.69 KByte/s | 10% |
+| 10/10 'yes' | 0.85% | 0.2% | 114.09 KByte/s | 0% |
+| no filter | 0.85% | 0.2% | 114.05 KByte/s | Baseline |
+
+![improvement_cpu](../img/content_filter/improvement_cpu.png)
+![improvement_memory](../img/content_filter/improvement_memory.png)
+![improvement_tx](../img/content_filter/improvement_tx.png)
+
+**The bigger data it handles, the more improvement we can have.**
+
+### Specification
+
+#### DDS 
+
+the following create/delete API's are defined,
+
+- create_contentfilteredtopic()
+- delete_contentfilteredtopic()
+
+**According to the specification, filter_expression can be only initialized at constructor and it is read_only.
+Probably it is not possible to change the filter_expression at runtime based on the specification but [RTI v5.3.0](https://community.rti.com/static/documentation/connext-dds/5.3.0/doc/api/connext_dds/api_cpp2/classdds_1_1topic_1_1ContentFilteredTopic.html#a76310bf0b7123dd89afbacf43dbabf4a) can support this requirement.

```

Comment:
filter_expression can be changed at runtime
If the spec is unclear you could consider filing an issue https://issues.omg.org/issues/create-new-issue?specification=DDS%231.4

Reply:
@mitza-oci thanks for the info. this was kinda confirmation since i was not clear on this.